lettuce/django/steps/models.py

```diff
@@ -199,7 +220,40 @@ def _dump_model(model, attrs=None):
     print
 
 
-def models_exist(model, data, queryset=None):
+def test_existence(model_or_queryset, data):
+    """
+    Test existence of a given hash in a queryset (or among all model instances
+    if a model is given).
+    """
+
+    try:
+        queryset = model_or_queryset.objects
+    except AttributeError:
+        queryset = model_or_queryset
+
+    fields = {}
+    extra_attrs = {}
+    for k, v in data.iteritems():
+        if k.startswith('@'):
+            # this is an attribute
+            extra_attrs[k[1:]] = v
+        else:
+            fields[k] = v
+
+    filtered = queryset.filter(**fields)
+
+    if filtered.exists():
+        for obj in filtered.all():
+            if all(getattr(obj, k) == v
+                    for k, v in extra_attrs.iteritems()):
```

Comment:
`k`, `v` isn't very descriptive... I found this block a little hard to read. I think that it could be nicer if the vars were called something better maybe... they aren't just keys/values, they're fields that you're searching for values in